.github/README.md

```diff
@@ -0,0 +1,126 @@
+
+  TESTING_RUST_VERSION: nightly-2024-09-05
+  TESTING_RUST_CACHE_KEY: "testing"
+
+  DEV_RUST_CACHE_KEY: "dev"
+
+  CLIPPY_RUST_CACHE_KEY: "clippy"
+```
+
+These variables define:
+
+- Versions for build and testing tools
+- Cache keys for various environments
+- Prefixes and other configuration values
+
+Having different cache for build environments allows to reduce the total cache size while still increasing speed of builds.
+
+## Running Workflows Locally with Act
+
+[Act](https://github.com/nektos/act) is a tool that allows you to run GitHub Actions workflows locally. This is useful for testing and debugging workflows without having to push changes to GitHub.
+
+**Warning**: Currently `act` doesn't work well with multiple level of dockerization. Meaning workflows which are using `uses: task-...` as jobs (ex: `pull-request-...` or `push-...`) won't work. However all the `task-...` workflows should work properly.
+
+### Common usages
+
+- `-s GITHUB_TOKEN="$(gh auth token)"` to setup the token
+- `--artifact-server-path ./tmp/artifacts/` to maintain artifacts used across tasks
+- `-P karnot-arc-runner-set=catthehacker/ubuntu:rust-22.04` allows task using `runs-on: karnot-arc-runner-set` to execute using act's ubuntu image with rust preinstalled
+
+```bash
+# Run the main build task (takes ~10min on good computer)
+act -W .github/workflows/task-build-madara.yml -s GITHUB_TOKEN="$(gh auth token)"  --artifact-server-path ./tmp/artifacts/ workflow_call -P karnot-arc-runner-set=catthehacker/ubuntu:rust-22.04
+
+# Output about the binaries hashes:
+# | madara-binary-hash: c2098b181b6d5d54fcb4a23df9df933d0b842fd615b00b7f2ef5e5783686b7b4
+# | orchestrator-binary-hash: ce354e4078a0a9ff2303e50255415f54edde012151edbd650e42187e91a8f0c1
+# | cairo-artifacts-hash: 07660f3d160f65ee5c66ea05bfd3f5ccc35ed1b4589bffc2230f2c798f0a5c15
+
+# Run the javascript tests (using --input to pass the binaries)
+act -W .github/workflows/task-test-js.yml -s GITHUB_TOKEN="$(gh auth token)"  --artifact-server-path ./tmp/artifacts/ workflow_call --input madara-binary-hash="c2098b181b6d5d54fcb4a23df9df933d0b842fd615b00b7f2ef5e5783686b7b4" --input cairo-artifacts-hash="07660f3d160f65ee5c66ea05bfd3f5ccc35ed1b4589bffc2230f2c798f0a5c15"
+
+# Run Orchestrator Coverage (using -s for passing secrets)
+act -W .github/workflows/task-coverage-orchestrator.yml -s GITHUB_TOKEN="$(gh auth token)" --artifact-server-path ./tmp/artifacts/ workflow_call -P karnot-arc-runner-set=catthehacker/ubuntu:rust-22.04 -s ETHEREUM_SEPOLIA_BLAST_RPC="https://sepolia.blast.io" -s RPC_FOR_SNOS="..."
+
+# Independantly build the production docker image used in the release process
+docker build -f docker/madara-production.Dockerfile . --build-arg="PYTHON_VERSION=3.9" --build-arg="SCARB_VERSION=2.8.2" --build-arg=FOUNDRY_VERSION=nightly
+```
+
+### Troubleshooting
+
+If you encounter issues:
+
+1. Check if Act is using the correct Docker images
+2. Ensure all required secrets are provided
```

Comment:
Where would we get the list of required secrets? From `.github/config/env.yml`?

Reply:
Well, they are secrets so you need to find the right person how has them :p

Reply:
![image](https://github.com/user-attachments/assets/429700c1-518f-40d9-a628-361d691081b3)